dataclients/kubernetes/clusterstate.go

```diff
@@ -55,6 +51,26 @@ func (state *clusterState) GetEndpointsByService(namespace, name, protocol strin
 		TargetPort: servicePort.TargetPort.String(),
 	}
 
+	if cached, ok := state.cachedEndpoints[epID]; ok {
+		return cached
+	}
+
+	ep, ok := state.endpoints[epID.ResourceID]
+	if !ok {
+		return nil
+	}
+
+	targets := ep.targetsByServicePort(protocol, servicePort)
+	sort.Strings(targets)
+	state.cachedEndpoints[epID] = targets
```

Comment:
This does not look correct, either mutex should be used or cache should not be updated

Reply:
fixed